src/metadata-parser.h

```diff
@@ -120,7 +120,7 @@ namespace librealsense
             auto s = reinterpret_cast<const S*>(((const uint8_t*)frm.additional_data.metadata_blob.data()) + _offset);
 
             if (!is_attribute_valid(s))
-                throw invalid_value_exception("metadata not available");
+                throw invalid_value_exception("metadata not available (to enable metadata support in your OS, see the documentation)");
```

Comment:
@radfordi , there is a nuance - the metadata is not guaranteed for all frames even if it is enabled on OS level as it is the FW that controls it. Hence the absence of metadata attribute shouldn't be automatically attributed to OS.
I'm really looking how to make this message informative yet concise as it is expected that this will be fired more than once per streaming session. Do you have any suggestion?

Reply:
The way to make this message most informative is to detect if support is present in the Linux kernel, or if the appropriate registry magic has been done on Windows, and to change the message based on the result.  In the meantime, I think my suggestion in this patch is a significant improvement to the status quo.

It's a large roadblock to expect user's to know via some magic, since it's not CAPITAL LETTERS at to top of the README, nor even in the README *at all*, that you'll likely have to patch your kernel to get auto-exposure times.  There's no warning in the `realsense-viewer` that you are using an unpatched kernel.  There is *zero* feedback.  It was a hair pulling experience for me to get exposure times and it's clearly the case for *many* other users.  In the end I had to talk to someone on the firmware team to find out that I had to upgrade my firmware and patch my kernel before this basic functional would work.  Providing feedback to users in the first message they will see is the least we can do.